packages/headless/src/features/facets/category-facet-set/category-facet-set-analytics-actions.ts

```diff
@@ -0,0 +1,68 @@
+import {ArrayValue} from '@coveo/bueno';
+import {CategoryFacetMetadata} from 'coveo.analytics/src/searchPage/searchPageEvents';
+import {SearchAppState} from '../../../state/search-app-state';
+import {validatePayload} from '../../../utils/validate-payload';
+import {
+  AnalyticsType,
+  makeAnalyticsAction,
+} from '../../analytics/analytics-utils';
+import {
+  buildFacetStateMetadata,
+  getStateNeededForFacetMetadata,
+} from '../facet-set/facet-set-analytics-actions-utils';
+import {
+  facetIdDefinition,
+  requiredNonEmptyString,
+} from '../generic/facet-actions-validation';
+
+interface CategoryFacetBreadcrumbPayload {
+  categoryFacetId: string;
+  categoryFacetPath: string[];
+}
+
+const categoryFacetBreadcrumbPayloadDefinition = {
+  categoryFacetId: facetIdDefinition,
+  categoryFacetPath: new ArrayValue({
+    required: true,
+    each: requiredNonEmptyString,
+  }),
+};
+
+const getCategoryFacetMetadata = (
+  state: Partial<SearchAppState>,
+  {categoryFacetId, categoryFacetPath}: CategoryFacetBreadcrumbPayload
+): CategoryFacetMetadata => {
+  const facet = state.categoryFacetSet![categoryFacetId]!;
+  const categoryFacetField = facet?.field;
+  const categoryFacetTitle = `${categoryFacetField}_${categoryFacetId}`;
+  return {
+    categoryFacetId,
+    categoryFacetPath,
+    categoryFacetField,
+    categoryFacetTitle,
```

Comment:
I think we can reuse `buildFacetBaseMetadata` here and add just the `categoryFacetPath` parameter?

Reply:
All keys are different unfortunately